samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -65,7 +65,7 @@
   public ZkJobCoordinator(Config config) {
     this.config = config;
     ZkConfig zkConfig = new ZkConfig(config);
-    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getAppId());
+    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getGlobalAppId());
```

Comment:
Yeah. That's what I thought. Reading the code, it seems like appId is aka jobId.